replica/database.hh

```diff
@@ -686,21 +686,24 @@ public:
     //    reader and a _bounded_ amount of writes which arrive later.
     //  - Does not populate the cache
     // Requires ranges to be sorted and disjoint.
+    // When compaction_time is engaged, the reader's output will be compacted, with the provided query time.
+    // This compaction doesn't do tombstone garbage collection.
```

Comment:
Well, we should. Perhaps in a separate series.

Reply:
This was a conscious decision. See the huge discussion on https://github.com/scylladb/scylladb/issues/3561 for more context.
TLDR: @vladzcloudius insists that we should do all we can do to prevent data resurrection, this includes sending expired tombstones. If we have it, we might as well send it as the remote could have data it covers.

Reply:
> That's why we have tombstone_gc = repair.

As mentioned in the referenced discussion, this doesn't solve all the problems and hence we should continue sending expired tombstones.
@avikivity note that this PR deals with the sending part of streaming (hence "before" in the name).

Expired tombstones can and should be dropped on the receiving side of the streaming. It think some work has already been done on the topic by @asias.

Bottom line: this part of the PR is good as-is and IMO no need to change that in the future.

Reply:
An expired tombstone could just as well have been dropped a minute before streaming started due to compaction. It's not reliable. If there are problems with resurrection, they should be solved; streaming expired tombstones doesn't solve anything.